src/main/java/de/neuland/jade4j/JadeConfiguration.java

```diff
@@ -44,22 +45,30 @@ public JadeConfiguration() {
             MAX_ENTRIES + 1).build();
 
     public JadeTemplate getTemplate(String name) throws IOException, JadeException {
+        return getTemplate(name, null, Collections.EMPTY_MAP);
+    }
+
+    public JadeTemplate getTemplate(String name, String languageCode, Map<String, String> translation) throws IOException, JadeException {
         if (caching) {
             long lastModified = templateLoader.getLastModified(name);
-            String key = name + "-" + lastModified;
-
+            String key;
+            if (languageCode != null) {
+                key = name + "-" + languageCode + "-" + lastModified;
+            } else {
+                key = name + "-" + lastModified;
```

Comment:
for the null case, it might be better to use the key
"template-<null>-123"

otherwise, if we have templates `template` and `template-FR` for whatever reason then the cache keys will clash.